specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/Disks.json

```diff
@@ -0,0 +1,319 @@
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskIdParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The disk has been returned.",
+            "schema": {
+              "$ref": "Disks.json#/definitions/Disk"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DiskProperties": {
+      "description": "Managed disk properties.",
+      "type": "object",
+      "properties": {
+        "diskId": {
+          "description": "The disk id.",
+          "type": "string",
+          "readOnly": false
+        },
+        "status": {
+          "description": "The disk status.",
+          "$ref": "Disks.json#/definitions/DiskState",
+          "readOnly": false
+        },
+        "sharePath": {
+          "description": "The disk share path.",
+          "type": "string",
+          "readOnly": false
+        },
+        "actualSize": {
+          "description": "The actual size of disk.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": false
+        },
+        "managedBy": {
+          "description": "Compute resource Uri which owns this disk.",
+          "type": "string",
+          "readOnly": false
+        },
+        "tenantId": {
+          "description": "The disk resource Uri from tenant view.",
+          "type": "string",
+          "readOnly": false
+        },
+        "resourceType": {
+          "description": "The type of the resource.",
+          "$ref": "Disks.json#/definitions/DiskResourceType",
+          "readOnly": false
+        },
+        "diskSku": {
+          "description": "the disk sku.",
+          "$ref": "Disks.json#/definitions/AccountType",
+          "readOnly": false
+        }
+      }
+    },
+    "Disk": {
+      "description": "Managed Disk.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "Disks.json#/definitions/DiskProperties",
+          "description": "Disk properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../2015-12-01-preview/Compute.json#/definitions/Resource"
+        }
+      ]
+    },
+    "DiskList": {
+      "type": "object",
+      "description": "List of disks.",
+      "properties": {
+        "value": {
+          "description": "List of disks.",
+          "type": "array",
+          "items": {
+            "$ref": "Disks.json#/definitions/Disk"
+          }
+        },
+        "nextLink": {
+          "description": "URI to the next page.",
+          "type": "string"
+        }
+      }
+    },
+    "DiskState": {
+      "description": "Disk State.",
+      "type": "string",
+      "enum": [
+        "Undefined",
```

Comment:
Can this ever be returned to a user?

Reply:
Yes, it will be return to user.

Reply:
Yes, it might return to user if it is in that state in server side. I would keep it as the same as what server side has.